torchgen/model.py

```diff
@@ -946,6 +978,33 @@ def native_function_class_name(self) -> Optional[str]:
             # updating every kernel definition + callsite of every in-tree aten kernel.
             return None
 
+    def should_gen_dispatchless_composite(self, f: NativeFunction) -> bool:
+        if f.func.name in self.index:
+            return False
+        return len(f.composite) > 0 and self.dispatch_key in COMPOSITE_DISPATCH_KEYS
+
+    def dispatchless_composite_struct(self, f: NativeFunction) -> str:
+        name = f.func.name.unambiguous_name()
+        return f"{name}_{self.dispatch_key.lower()}"
+
+    def dispatchless_composite_kernel(self, f: NativeFunction) -> str:
+        struct_name = self.dispatchless_composite_struct(f)
+
+        # Similar to cpp.name().
+        # Couldn't use it due to circular dependency issues.
+        name = str(f.func.name.name)
+        if f.func.is_out_fn():
+            name += "_out"
+
+        return f"{name}<{struct_name}>"
```

Comment:
These feel like they should belong in torchgen.api, not here

Reply:
Maybe a workaround would be to `from torchgen.api import cpp` inside this function. Would that be any better?

Reply:
yes, that's fine, though I think my suggestion is to move the entire API into torchgen.api

Reply:
+1 on moving this out of `model.py`